cloudinit/sources/DataSourceNoCloud.py

```diff
@@ -37,15 +39,17 @@ def __str__(self):
         return "%s [seed=%s][dsmode=%s]" % (root, self.seed, self.dsmode)
 
     def _get_devices(self, label):
-        fslist = util.find_devs_with("TYPE=vfat")
-        fslist.extend(util.find_devs_with("TYPE=iso9660"))
-
-        label_list = util.find_devs_with("LABEL=%s" % label.upper())
-        label_list.extend(util.find_devs_with("LABEL=%s" % label.lower()))
-        label_list.extend(util.find_devs_with("LABEL_FATBOOT=%s" % label))
-
-        devlist = list(set(fslist) & set(label_list))
-        devlist.sort(reverse=True)
+        if platform.system().lower() == "aix":
+            devlist = aix_util.find_devs_with("cd0")
```

Comment:
i know the structure in util.py isn't great, but, why aren't you extending that, rather tumač touching every DataSource separately?